src/flags.rs

```diff
@@ -326,13 +326,23 @@ pub fn init_flag_array(
     })
 }
 
-/// Expand a flag array of `[timestep][channel][baseline]` to `[timestep][channel][baseline][pol]`
-pub fn expand_flag_array(flag_array: ArrayView3<bool>, num_pols: usize) -> Array4<bool> {
-    let shape = flag_array.dim();
-    Array4::from_shape_fn(
-        (shape.0, shape.1, shape.2, num_pols),
-        |(ts_idx, ch_idx, bl_idx, _)| flag_array[[ts_idx, ch_idx, bl_idx]],
-    )
+/// Expand an array into a new axis by repeating each element `size` times
+pub fn add_dimension<T>(array: ArrayView3<T>, size: usize) -> Array4<T>
+where
+    T: std::clone::Clone,
+{
+    let shape = array.dim();
+    array
+        .insert_axis(Axis(3))
+        .broadcast((shape.0, shape.1, shape.2, size))
+        .unwrap()
+        .to_owned()
+}
+
+/// Expand a flag array into a new axis by repeating each element `size` times
+#[deprecated(since = "0.4.1", note = "please use `add_dimension` instead")]
+pub fn expand_flag_array(array: ArrayView3<bool>, size: usize) -> Array4<bool> {
```

Comment:
@cjordan if we deprecate a function, are we considering that enough of a breaking API change to need to release a new minor version? Considering just releasing this as birli 0.4.1

Reply:
I'm not sure! I think you should release 0.4.1 anyway as we're likely to have a 0.5 in the near future (new Marlu etc.)